src/control/grouphandle.cpp

```diff
@@ -63,19 +122,56 @@ int resetAllGroupHandles() {
     tmpGroupHandlesByName.swap(allGroupHandlesByName);
     allGroupHandlesFrozen.storeRelease(0);
     locked.unlock();
-    for (const auto groupHandle : qAsConst(tmpGroupHandlesByName)) {
+    for (const auto& groupHandle : qAsConst(tmpGroupHandlesByName)) {
         qInfo() << "Deleting" << groupHandle;
-        delete groupHandle;
+        delete groupHandle.m_pImpl;
```

Comment:
I'm not comfortable with manually deleting objects out from someone else. Shouldn't instead take the group_handle take care of deleting `m_pImpl` and we just call [`QHash::erase`](https://doc.qt.io/qt-5/qhash.html#erase) on each groupHandle?

Reply:
Not possible because `m_pImpl` is shared by many instances.

Reply:
So that results in a use-after-free then?! Or can we guarantee that it won't be accessed anymore after this delete call?

Reply:
No guarantees. This method must be invoked after all handles have already been dropped and is only provided for testing purposes and to make the leak checker happy.

Reply:
-> Used in tests

Reply:
Mhmm ok, makes sense. I'll add a comment.